src/share/poudriere/image_firmware.sh

```diff
@@ -96,7 +99,7 @@ firmware_build()
 	OS_SIZE=
 	calculate_ospart_size "2" "${NEW_IMAGESIZE_SIZE}" "${CFG_SIZE}" "${DATA_SIZE}" "${SWAPSIZE}"
 	# Prune off a bit to fit the extra partitions and loaders
-	OS_SIZE=$(( $OS_SIZE - 1 ))
+	OS_SIZE=$(( OS_SIZE - 1 - ESP_SIZE / 2 ))
```

Comment:
Another option would be to define the `ESP_SIZE` to 1MB (closer to the original 800KB) and avoid this calculation, however having a larger ESP may have its uses (BIOS firmware upgrades, etc.).